vital/types/detected_object.h

```diff
@@ -76,9 +79,18 @@ class VITAL_EXPORT detected_object
 {
 public:
 
-  typedef std::vector< detected_object_sptr > vector_t;
-  typedef descriptor_dynamic< double > descriptor_t;
-  typedef std::shared_ptr< descriptor_t const > descriptor_scptr;
+  using vector_t = std::vector< detected_object_sptr >;
+  using descriptor_t = descriptor_dynamic< double >;
+  using descriptor_scptr = std::shared_ptr< descriptor_t const >;
+
+  /**
+   * @brief Create default detected object.
+   *
+   * @param confidence Detectors confidence in this detection.
+   * @param classifications Optional object classification.
+   */
+  detected_object( double confidence = 1.0,
+                   detected_object_type_sptr classifications = nullptr);
```

Comment:
Nit: missing space inside `)` (also next two ctors)... apologies for not noticing before.